src/state_machine.rs

```diff
@@ -245,35 +266,33 @@ impl Debug for StateMachine {
     }
 }
 
-#[cfg(not(feature = "use-mock-crust"))]
 impl StateMachine {
-    /// Run the event loop for sending and receiving messages. Blocks until
-    /// the core is terminated, so it must be called in a separate thread.
-    pub fn run(&mut self) {
-        while self.is_running {
+    #[cfg(not(feature = "use-mock-crust"))]
```

Comment:
Why limit this method to the production code? The only difference to `try_step` I see is that it will block the thread - I'm not sure if that will ever be useful in tests, but I can't see any reason to ban this altogether.